### ISSUE TITLE
Feature request: Snapshots using thick LVM storage pools.

### ISSUE THREAD

**Issue by space-p** [LINBIT/linstor-server#148]
hello,

Sometimes, we need to use thick LVM, for maximum performance.
It would be great to be able to create snapshots , this necessary for consistent backup of block devices without freezing IO.

additional parameter: COW table size (lvm example: lvcreate -s -nsnap1 -l50%ORIGIN vg/origin).

**Comment by ghernadi** [comment]
To cite a colleague: "Agreed, it **would** be great" :)

The problem with thick snapshots is actually performance. Please try it yourself, simply create a bunch of snapshots from a thin volume as well as from a thick volume and try to write on the original volume afterwards. Take care to measure correctly to not let caches invalidate your results.

My tests showed that writes to thick volumes with a few snapshots already take about twice as long as to thin volumes.

As you figured, LVM already supports it. Performance is actually the only reason we decided to "hide" this LVM feature in order to not confuse users why the writes are suddenly so slow.

**Comment by space-p** [comment]
Yeah, that's right, the write performance is halved for each snapshot of a thick LV.
However, the profile of using snapshots on a thick LV differs from the profile on a thin LV.  It is needed in most cases for a consistent backup and the snapshot's lifetime is the time it took to create a backup from the snapshot(create_snap->backup->remove_snap).

Can I use the hidden feature?
I don’t want to create snapshot LV bypassing linstore :(

**Comment by ghernadi** [comment]
Even if you have "short" lifetimes for those snapshots, during the backuping process you will have bad write performance on the actual volume. Not sure how large your volumes are or how fast your network / "backuping-speed" is, but I guess you can do the calculations by yourself what timeframe we are talking about.

Right now, there is no option in Linstor to make snapshots of thick LVM volumes.

**Comment by space-p** [comment]
But why not use this option ... with a note that write I/O performance will be halved for every snapshot?
This option will allow us to create a snapshot directly from linstor controller (not bypass).
